src/overlays/actors/ovl_En_Horse_Ganon/z_en_horse_ganon.c

```diff
@@ -25,29 +45,291 @@ const ActorInit En_Horse_Ganon_InitVars = {
+    this->actor.speedXZ = 0.0f;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void func_80A68B20(EnHorseGanon* this) {
+    s32 animationChanged;
+    f32 sp30;
+
+    animationChanged = 0;
+    this->updateFnIndex = 1;
+    if (this->actor.speedXZ <= 3.0f) {
+        if (this->currentAnimation != 2) {
+            animationChanged = 1;
+        }
+        this->currentAnimation = 2;
+    } else if (this->actor.speedXZ <= 6.0f) {
+        if (this->currentAnimation != 3) {
+            animationChanged = 1;
+        }
+        this->currentAnimation = 3;
+    } else {
+        if (this->currentAnimation != 4) {
+            animationChanged = 1;
+        }
+        this->currentAnimation = 4;
+    }
+
+    if (this->currentAnimation == 2) {
+        sp30 = this->actor.speedXZ / 3.0f;
+    } else if (this->currentAnimation == 3) {
+        sp30 = this->actor.speedXZ / 5.0f;
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_RUN, &this->actor.unk_E4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    } else if (this->currentAnimation == 4) {
+        sp30 = this->actor.speedXZ / 7.0f;
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_RUN, &this->actor.unk_E4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    } else {
+        sp30 = 1.0f;
+    }
+
+    if (animationChanged == 1) {
+        SkelAnime_ChangeAnim(&this->skelAnime, D_80A691B0[this->currentAnimation],
+                             animPlaybackSpeed[this->currentAnimation] * sp30 * 1.5f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_80A691B0[this->currentAnimation]->genericHeader), 2, -3.0f);
+    } else {
+        SkelAnime_ChangeAnim(&this->skelAnime, D_80A691B0[this->currentAnimation],
+                             animPlaybackSpeed[this->currentAnimation] * sp30 * 1.5f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_80A691B0[this->currentAnimation]->genericHeader), 2, 0.0f);
+    }
+}
+
+void func_80A68DB0(EnHorseGanon* this, GlobalContext* globalCtx) {
+    if (this->currentAnimation == 2) {
+        func_80A68870(this);
+    }
+
+    func_80A686A8(this, globalCtx);
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        func_80A68B20(this);
+    }
+}
+
+void func_80A68E14(EnHorseGanon* this, GlobalContext* globalCtx) {
+    u32 junk; // aligns stack properly
+    CollisionPoly* col;
+    f32 temp_ret;
+    Vec3f v;
+    s32 temp1;
+
+    v.x = Math_Sins(this->actor.shape.rot.y) * 30.0f + this->actor.posRot.pos.x;
+    v.y = this->actor.posRot.pos.y + 60.0f;
+    v.z = Math_Coss(this->actor.shape.rot.y) * 30.0f + this->actor.posRot.pos.z;
+
+    temp_ret = func_8003C940(&globalCtx->colCtx, &col, &temp1, &v);
+
+    this->unk_1f4 = temp_ret;
+    this->actor.shape.rot.x = D_80A692D0 * Math_atan2f(this->actor.posRot.pos.y - temp_ret, 30.0f);
+}
+
+#ifdef NON_MATCHING
+void EnHorseGanon_Update(EnHorseGanon* this, GlobalContext* globalCtx)
+// simply needs to store a0 in s0 a few instructions earlier
+{
+    u8 padding[8];
```

Comment:
we generally leave stack padding as words unless needed. its not a big deal but might as well make this an array of 2 s32's